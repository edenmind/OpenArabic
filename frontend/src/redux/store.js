import {
  arabicSentenceReducer,
  arabicTextReducer,
  arabicWordsReducer,
  authorReducer,
  categoryReducer,
  englishSentenceReducer,
  englishTextReducer,
  englishWordsReducer,
  sourceReducer,
  textReducer,
  wordByWordReducer,
} from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    title: textReducer,
    category: categoryReducer,
    author: authorReducer,
    source: sourceReducer,
    englishText: englishTextReducer,
    arabicText: arabicTextReducer,
    arabicWords: arabicWordsReducer,
    englishWords: englishWordsReducer,
    arabicSentence: arabicSentenceReducer,
    englishSentence: englishSentenceReducer,
    wordByWord: wordByWordReducer,
  },
})
