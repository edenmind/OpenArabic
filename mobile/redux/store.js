import { configureStore, combineReducers } from '@reduxjs/toolkit'

import { categoriesReducer, textReducer, textsReducer, UIStateReducer, wordsReducer } from './reducers.js'

const rootReducer = combineReducers({
  categories: categoriesReducer,
  text: textReducer,
  texts: textsReducer,
  textLoading: UIStateReducer,
  textsLoading: UIStateReducer,
  arabicFontSize: UIStateReducer,
  englishFontSize: UIStateReducer,
  isTransliterationOn: UIStateReducer,
  arabicFontName: UIStateReducer,
  practicingWords: UIStateReducer,
  words: wordsReducer,
  isDarkMode: UIStateReducer
})

const store = configureStore({
  reducer: rootReducer
})

export { store }
