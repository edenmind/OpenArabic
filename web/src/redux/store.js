import * as reducers from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    text: reducers.textReducer,
  },
})
