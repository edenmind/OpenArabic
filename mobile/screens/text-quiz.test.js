/* eslint-disable putout/objects-braces-inside-array */
import { expect, describe, it, jest } from '@jest/globals'
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import { store } from '../redux/store.js'
import TextQuiz from './text-quiz.js'

jest.useFakeTimers()
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

jest.mock('../components/spinner.js')

//mock useSelector by returning a store with a text
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: () => ({
    text: {
      textLoading: true,
      text: {
        arabic: ['a1', 'a2', 'a3'],
        english: ['e1', 'e2', 'e3'],
        status: 'Draft',
        title: 'abc',
        publishAt: '2021-01-01T00:00:00.000Z',
        wordByWord: [['']],
        image: 'abc',
        texts: { arabic: '', english: '' },
        category: 'abc',
        author: '',
        arabicSentence: [''],
        source: '',
        vocabularyCollection: {
          numberOfBatches: 1,
          arabic: [[{ word: 'a', wordId: '1' }]],
          english: [[{ word: 'e', wordId: '1' }]]
        },
        // eslint-disable-next-line putout/objects-braces-inside-array
        sentences: [
          {
            english: 'e1',
            arabic: 'a1',
            words: [
              {
                arabic: 'a',
                english: '1'
              }
            ]
          },
          {
            english: 'e2',
            arabic: 'a2',
            words: [
              {
                arabic: 'a',
                english: '2'
              }
            ]
          },
          {
            english: 'e3',
            arabic: 'a3',
            words: [
              {
                arabic: 'a',
                english: '3'
              }
            ]
          }
        ]
      }
    }
  })
}))

it('renders correctly', () => {
  const rendered = renderer
    .create(
      <Provider store={store}>
        <TextQuiz />
      </Provider>
    )
    .toJSON()
  expect(rendered).toMatchSnapshot()
})
