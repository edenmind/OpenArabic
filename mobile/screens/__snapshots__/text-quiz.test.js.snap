// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "paddingTop": 25,
          "width": "50%",
        }
      }
    >
      <View
        backgroundColor="#a4cfbe"
        collapsable={false}
        style={
          {
            "backgroundColor": "#e4f2d6",
            "borderColor": "#e4f2d6",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "direction": "rtl",
            "elevation": 0,
            "flexDirection": "column",
            "height": 50,
            "margin": 15,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "borderRadius": 16,
                },
                {
                  "flexGrow": 1,
                },
              ],
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "paddingLeft": 4,
                  "position": "relative",
                },
                {
                  "paddingRight": 4,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "lineHeight": 24,
                      "marginVertical": 4,
                      "minHeight": 24,
                      "textAlignVertical": "center",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "marginLeft": 8,
                      "marginRight": 8,
                    },
                    {
                      "fontSize": 25,
                    },
                  ],
                ]
              }
            >
              a
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "paddingTop": 25,
          "width": "50%",
        }
      }
    >
      <View
        backgroundColor="#a4cfbe"
        collapsable={false}
        style={
          {
            "backgroundColor": "#e4f2d6",
            "borderColor": "#e4f2d6",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "direction": "ltr",
            "elevation": 0,
            "flexDirection": "column",
            "height": 50,
            "margin": 15,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "borderRadius": 16,
                },
                {
                  "flexGrow": 1,
                },
              ],
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "paddingLeft": 4,
                  "position": "relative",
                },
                {
                  "paddingRight": 4,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "lineHeight": 24,
                      "marginVertical": 4,
                      "minHeight": 24,
                      "textAlignVertical": "center",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "marginLeft": 8,
                      "marginRight": 8,
                    },
                    {
                      "fontSize": 16,
                    },
                  ],
                ]
              }
            >
              e
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  />,
]
`;
